src/expr.rs

```diff
@@ -2101,6 +2101,25 @@ pub(crate) fn rewrite_assign_rhs_expr<R: Rewrite>(
     rhs_kind: &RhsAssignKind<'_>,
     rhs_tactics: RhsTactics,
 ) -> Option<String> {
+    let get_lhs_last_line_shape = || {
+        let last_line = lhs.rsplitn(2, "\n").next().unwrap_or_default();
+        let tab_spaces = context.config.tab_spaces();
+        let new_shape = shape
+            .block_indent(tab_spaces)
+            .saturating_sub_width(tab_spaces);
+        let extra_indent_string = new_shape.to_string(&context.config).to_string();
+        if last_line.starts_with(&extra_indent_string) {
```

Comment:
We shouldn't need to allocate a string with `to_string()`. We should be able to call `as_ref()`:

```suggestion
        let extra_indent_string = new_shape.to_string(&context.config);
        if last_line.starts_with(extra_indent_string.as_ref()) {
```